packages/webview_flutter/lib/platform_interface.dart

```diff
@@ -0,0 +1,44 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+
+import 'package:flutter/foundation.dart';
+import 'package:flutter/gestures.dart';
+import 'package:flutter/widgets.dart';
+
+/// Interface for talking to the webview controller on the platform side.
+/// 
+/// An instance implementing this interface is passed to the `onWebViewCreated` callback that is
+/// passed to [WebViewPlatformInterface#onWebViewCreated].
+abstract class WebViewPlatformControllerInterface {
```

Comment:
Actually I think we better make this class non-abstract and have a default implementation(that throws?) for all methods, this way a new implementor doesn't have to implement the full API surface all at once (and when we're adding new methods here it's going to take time for implementors to catch-up).

I'll update the PR to make this non abstract tomorrow.

Reply:
I  think it can stay abstract but you can just add default implementation which others will inherit when extending it

Reply:
what @nkoroste says

Reply:
ok, added a default implementation that throws.

Reply:
If the default implementation throws, it's probably best if the default is abstract (so that the analyzer tells you to implement it). Otherwise you risk crashes at runtime and there's no lint to tell you what you need to implement. If it makes sense to have an actual default that doesn't throw, I'd go with that. That way you can get up and going quickly on a new platform.